tests/backup/backup_helper.go

```diff
@@ -1242,7 +1246,377 @@ func backupSuccessCheck(backupName string, orgID string, retryDuration time.Dura
+	log.InfoD("Validating backup [%s] in org [%s]", backupName, orgID)
+
+	log.InfoD("Obtaining backup info for backup [%s]", backupName)
+	backupDriver := Inst().Backup
+	backupUid, err := backupDriver.GetBackupUID(ctx, backupName, orgID)
+	if err != nil {
+		return nil, fmt.Errorf("GetBackupUID Err: %v", err)
+	}
+	backupInspectRequest := &api.BackupInspectRequest{
+		Name:  backupName,
+		Uid:   backupUid,
+		OrgId: orgID,
+	}
+	backupInspectResponse, err := backupDriver.InspectBackup(ctx, backupInspectRequest)
+	if err != nil {
+		return nil, fmt.Errorf("InspectBackup Err: %v", err)
+	}
+
+	backupStatus := backupInspectResponse.GetBackup().GetStatus().Status
+	if backupStatus != api.BackupInfo_StatusInfo_Success &&
+		backupStatus != api.BackupInfo_StatusInfo_PartialSuccess {
+		return nil, fmt.Errorf("ValidateBackup requires backup [%s] to have a status of Success or PartialSuccess", backupName)
+	}
+
+	backedupAppContexts, ScheduledCtxNotFoundInBackupErrors, BackupNotFoundInAnyScheduledCtxErrors, otherErrors := GetBackupCtxsFromScheduledCtxs(backupInspectResponse, scheduledAppContexts)
+	// print all error, but don't consume them
+	ProcessMultipleErrors("GetBackupCtxsFromScheduledCtxs-ScheduledCtxNotFoundInBackupErrors", ScheduledCtxNotFoundInBackupErrors, true)
+	ProcessMultipleErrors("GetBackupCtxsFromScheduledCtxs-BackupNotFoundInAnyScheduledCtxErrors", BackupNotFoundInAnyScheduledCtxErrors, true)
+	ProcessMultipleErrors("GetBackupCtxsFromScheduledCtxs-otherErrors", otherErrors, true)
+
+	errArr := make([]error, 0)
+	if requireAllScheduledCtxAreInBackup {
+		errArr = append(errArr, ScheduledCtxNotFoundInBackupErrors...)
+	}
+	if requireAllBackupAreInSomeScheduledCtx {
+		errArr = append(errArr, BackupNotFoundInAnyScheduledCtxErrors...)
+	}
+	errArr = append(errArr, otherErrors...)
+	// consume required errors, but don't print
+	errGetBackupCtxsFromScheduledCtxs := ProcessMultipleErrors("GetBackupCtxsFromScheduledCtxs", errArr, false)
+
+	errors := ValidateBackedUpVolumes(backupInspectResponse, scheduledAppContexts)
+	errValidateBackedUpVolumes := ProcessMultipleErrors("ValidateBackedUpVolumes", errors, true)
+
+	if errGetBackupCtxsFromScheduledCtxs != nil || errValidateBackedUpVolumes != nil {
+		err = fmt.Errorf("GetBackupCtxsFromScheduledCtxs: %v ;; ValidateBackedUpVolumes: %v",
+			errGetBackupCtxsFromScheduledCtxs, errValidateBackedUpVolumes)
+	} else {
+		err = nil
+	}
+
+	return backedupAppContexts, err
+}
+
+// ValidateBackedUpVolumes checks if the volumes have been backed up and done so rightly
+// *
+// * Returns:
+// * otherErrors: all the other kind of error; Don't ignore these are they're serious errors
+// *
+// * NOTES:
+// * - make sure to include Contexts of *all* namespaces which are supposed to contain the volumes
+func ValidateBackedUpVolumes(backupInspectResponse *api.BackupInspectResponse,
```

Comment:
As part of volume validation, we will just check that volumes given by k8s API is present in the backup inspect response